packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/streamhandler/TransactionStreamHandler.java

```diff
@@ -148,9 +148,10 @@ public void onListen(Object arguments, EventSink events) {
               } else if (task.getResult() != null) {
                 map.put("complete", true);
               }
-
-              activityRef.get().runOnUiThread(() -> events.success(map));
-              activityRef.get().runOnUiThread(events::endOfStream);
+              if (activityRef.get() != null) {
+                activityRef.get().runOnUiThread(() -> events.success(map));
+                activityRef.get().runOnUiThread(events::endOfStream);
+              }
```

Comment:
```suggestion
              events.success(map);
              events.endOfStream();
```

Reply:
Doesn't seem necessary to `runOnUiThread` given it will run on the UI thread anyway according to Flutter [docs](https://docs.flutter.dev/get-started/flutter-for/android-devs#what-is-the-equivalent-of-runonuithread-in-flutter)

Reply:
Thank's for your suggest! I've removed unnecessary `activityRef`, can you please review my last changes?

Reply:
removing the `activityRef.get().runOnUiThread` gives an error every time, when i run a transaction

`[cloud_firestore/null] null`

Reply:
> Doesn't seem necessary to `runOnUiThread` given it will run on the UI thread anyway according to Flutter [docs](https://docs.flutter.dev/get-started/flutter-for/android-devs#what-is-the-equivalent-of-runonuithread-in-flutter)

This is the Dart/Flutter UI code, afaik this does not apply to plugin native code and that is ran off the UI thread.

I think the best solution here is to remove a requirement to hold the Activity entirely and just use the main looper directly like we do everywhere else, e.g. on RTDB transaction: https://github.com/FirebaseExtended/flutterfire/blob/eaf0af6c4b0d418aa36d7dd2cd8cd60e40b06ebf/packages/firebase_database/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/TransactionExecutor.java#L24-L26

Reply:
Hey @TimurDyushaliev, as suggested above, please remove `activityRef` & use the Handler implementation:
```android
new Handler(Looper.getMainLooper())
```
This will always provide the context of the main activity instead of checking if the activity exists and the possibility of missing an event.

Reply:
Yeah, this solution is great and works well. Thank you guys for helping me:) Can you please check my last updates? 

I hope I did as you mean and if not I'm open to refactor. And because it's my first ever contributing, I've also added my name to AUTHORS file as written in [Contributor Guide](https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md).